src/librustc_lint/context.rs

```diff
@@ -62,6 +63,9 @@ pub struct LintStore {
     /// Lints indexed by name.
     by_name: FxHashMap<String, TargetLint>,
 
+    /// Feature-gated lints.
+    gated_lints: FxHashMap<LintId, (String, Symbol)>,
```

Comment:
This doesn’t seem necessary to me? I think if you have a `LintId`, you can access the associated lint, and then check if the feature gate is enabled.

Reply:
Thanks, I missed that the `Lint` could be accessed that way. Fixed this in the latest commit and also moved `check_gated_lint` back to `levels.rs` where it was before.